packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx

```diff
@@ -51,20 +50,26 @@ const calculateDaysUp = (response: PrometheusResponse): number | null => {
     return null;
   }
 
-  const values: PrometheusValue[] = response.data?.result?.[0]?.values || [];
+  let minTimestamp: number;
+  let maxTimestamp: number = 0;
 
-  if (!values.length) {
-    return null;
-  }
+  for (const promResult of response.data.result) {
+    if (!promResult.values.length) {
+      // eslint-disable-next-line no-continue
+      continue;
+    }
 
-  const timestamps: number[] = values.map(([timestamp, _value]) => timestamp);
+    const timestamps: number[] = promResult.values.map(
+      ([timestamp, _value]) => timestamp
+    );
 
-  const minTimestamp: number = Math.min(...timestamps);
-  const maxTimestamp: number = Math.max(...timestamps);
+    minTimestamp = Math.min(...timestamps, minTimestamp ?? timestamps[0]);
+    maxTimestamp = Math.max(...timestamps, maxTimestamp);
+  }
 
-  const timeDifferenceMs: number = maxTimestamp - minTimestamp;
+  const timeDifferenceSecond: number = maxTimestamp - minTimestamp;
```

Comment:
for edge case if `promResult.values.length` is empty...
```suggestion
  const timeDifferenceSecond: number = maxTimestamp - (minTimestamp || 0);
```

Reply:
if length is zero, the total value will be 0 (0-null) is zero from what i can see on the console

Reply:
`0-undefined` is `NaN`

Reply:
most commonly u will find `undefined` with JS...

Reply:
updated